### PR TITLE
Implements the remaining features

### PR DESCRIPTION
**Pull request by filcarv**

### CODE REVIEW

File: Firmware/AnalogInput/app.c [hunk 1]
```diff
@@ -30,7 +30,7 @@ void hwbp_app_initialize(void)
     uint8_t hwH = 1;
     uint8_t hwL = 0;
     uint8_t fwH = 1;
-    uint8_t fwL = 0;
+    uint8_t fwL = 2;
```
Comment (bruno-f-cruz) [inline Firmware/AnalogInput/app.c:33]: Make sure to update the schema.yml to target this fw version

File: Firmware/AnalogInput/app_funcs.c [hunk 2]
```diff
@@ -312,6 +314,12 @@ void app_read_REG_DO0_PULSE(void)
 bool app_write_REG_DO0_PULSE(void *a)
 {
 	uint8_t reg = *((uint8_t*)a);
+	
+	if (reg < 1)
```
Comment (bruno-f-cruz) [inline Firmware/AnalogInput/app_funcs.c:318]: The minimum value in the schema should also be updated if I understand it correctly.

File: device.yml
Comment (bruno-f-cruz) [inline device.yml:1]: If you have implemented the digital output channels as outputs for the threshold operation, shouldn't the access in the schema of the respective addresses be changed from `private` to `public`?

**Review by bruno-f-cruz — CHANGES_REQUESTED** [review]

File: Firmware/AnalogInput/app_funcs.c (hunk 2 above)
Comment (filcarv) [reply Firmware/AnalogInput/app_funcs.c:318]: The minimum value was already 1. I don't think any change is needed.

File: Assets/test script.bonsai
Comment (bruno-f-cruz) [inline Assets/test script.bonsai:1]: Do we actually need this workflow? If so, can you give it a more descriptive name? ideally without spaces.

**Review by bruno-f-cruz — APPROVED** [review]
Looks good! Just check the issue with the testing workflow.

**Review by artursilva0 — APPROVED** [review]

File: Assets/test script.bonsai
Comment (filcarv) [reply Assets/test script.bonsai:1]: It was renamed to **HardwareUnitTests.bonsai**.

**Pull request merged**